dvc/config_schema.py

```diff
@@ -114,6 +114,7 @@ class RelPath(str):
         Optional("interactive", default=False): Bool,
         Optional("analytics", default=True): Bool,
         Optional("hardlink_lock", default=False): Bool,
+        "tmp_dir": str,
```

Comment:
Same problem as with state/index.dir - there is a chance that someone will try to share them between repos, which might get confusing (e.g. same lock for all repos, so they can no longer run in parallel). We could create a dir inside based on root_dir hash like state/index.dir do, but that will result in state being lost when a repo dir is moved, unless we generate the tmp_dir and then record it back somehow (either back into the config, or by creating a file with a new generated path inside, that we will then dereference). I don't like that we are trying to be too smart with these approaches, as it might bite us. The simple approach is probably better for now.